app/src/main/java/com/github/swent/echo/viewmodels/event/EventViewModel.kt

```diff
@@ -1,59 +1,81 @@
 package com.github.swent.echo.viewmodels.event
 
 import android.util.Log
+import androidx.lifecycle.SavedStateHandle
 import androidx.lifecycle.ViewModel
-import com.github.swent.echo.data.model.Association
+import androidx.lifecycle.viewModelScope
+import com.github.swent.echo.R
+import com.github.swent.echo.authentication.AuthenticationService
 import com.github.swent.echo.data.model.Event
-import com.github.swent.echo.data.model.Location
 import com.github.swent.echo.data.model.Tag
-import com.github.swent.echo.data.model.UserProfile
-import java.time.ZonedDateTime
+import com.github.swent.echo.data.repository.Repository
+import dagger.hilt.android.lifecycle.HiltViewModel
+import javax.inject.Inject
 import kotlinx.coroutines.flow.MutableStateFlow
 import kotlinx.coroutines.flow.asStateFlow
+import kotlinx.coroutines.launch
 
 /** represents an event, used in the event screens */
-class EventViewModel() : ViewModel() {
-    private val emptyEvent =
-        Event(
-            "",
-            UserProfile("", "", null, null, emptySet()),
-            Association("", "", ""),
-            "",
-            "",
-            Location("", 0.0, 0.0),
-            ZonedDateTime.now(),
-            ZonedDateTime.now(),
-            emptySet(),
-            0,
-            0,
-            0
-        )
-    private val _event = MutableStateFlow<Event>(emptyEvent)
-    private val _status = MutableStateFlow<EventStatus>(EventStatus.New)
+@HiltViewModel
+class EventViewModel
+@Inject
+constructor(
+    private val repository: Repository,
+    private val authenticationService: AuthenticationService,
+    private val savedEventId: SavedStateHandle
+) : ViewModel() {
 
-    // placeholder for the list of all tags
-    private val allTagsList = listOf(Tag("1", "tag1"), Tag("2", "tag2"), Tag("3", "tag3"))
+    private val _event = MutableStateFlow<Event>(Event.EMPTY)
+    private val _status = MutableStateFlow<EventStatus>(EventStatus.Modified)
+    val event = _event.asStateFlow()
+    val status = _status.asStateFlow()
+    private val allTagsList = MutableStateFlow<List<Tag>>(listOf())
```

Comment:
Does this needs to be a `MutableStateFlow` as it won't be exposed in any composable? This could simply be a list of tags (`private lateinit var allTags: List<Tag>`) which would be initialized in the `init` block.